Source/Actors/Player.cs

```diff
@@ -1,2406 +1,2448 @@
+        GetCameraTarget(out var orig, out var target, out _);
+        World.Camera.LookAt = target;
+        World.Camera.Position = orig;
+
+        if (Save.Instance.Freecam != Save.FreecamMode.Free)
+        {
+            Manager.FreeCamPosition = Position;
+        }
+
+        Manager.FreeCamDistance = freecamDistance;
+    }
+
+    public override void Update()
+    {
+        freecamPosition = Manager.FreeCamPosition;
+        freecamRotation = Manager.FreeCamRotation;
+        freecamDistance = Manager.FreeCamDistance;
+
+        if (Save.Instance.Freecam is Save.FreecamMode.Orbit or Save.FreecamMode.Free)
+        {
+            // Freecam rotation
+            if (Input.Mouse.Down(MouseButtons.Left))
+            {
+                //if the mouse moves too much, it means, the Game reset the mouse somewhere
+                //the camera would behave weirdly, if this was not there
+                if (World.MouseDelta.LengthSquared() > 10000) return;
+                freecamRotation.X += World.MouseDelta.X * Time.Delta * FreecamRotateSpeed;
+                freecamRotation.Y += World.MouseDelta.Y * Time.Delta * FreecamRotateSpeed;
+                freecamRotation.X %= 360.0f * Calc.DegToRad;
+                freecamRotation.Y = Math.Clamp(freecamRotation.Y, -89.9f * Calc.DegToRad, 89.9f * Calc.DegToRad);
+            }
+
+            // Freecam zoom
+            freecamDistance -= Input.Mouse.Wheel.Y * FreecamZoomSpeed;
+        }
+
+        if (Save.Instance.Freecam == Save.FreecamMode.Free)
+        {
+            // Freecam movement
+            var cameraForward = new Vec2(
+                MathF.Sin(freecamRotation.X),
+                MathF.Cos(freecamRotation.X));
+            var cameraRight = new Vec2(
+                MathF.Sin(freecamRotation.X - Calc.HalfPI),
+                MathF.Cos(freecamRotation.X - Calc.HalfPI));
+
+            // if not currently in TAS, proceed with normal movement
+            if (Manager.CurrState == Manager.State.Disabled)
+            {
+                freecamPosition.X -= Controls.Move.Value.Y * cameraForward.X * FreecamMoveSpeed;
+                freecamPosition.Y -= Controls.Move.Value.Y * cameraForward.Y * FreecamMoveSpeed;
+                freecamPosition.X -= Controls.Move.Value.X * cameraRight.X * FreecamMoveSpeed;
+                freecamPosition.Y -= Controls.Move.Value.X * cameraRight.Y * FreecamMoveSpeed;
+
+                if (Input.Keyboard.Down(Keys.Space))
+                    freecamPosition.Z += FreecamMoveSpeed;
+                if (Input.Keyboard.Down(Keys.LeftControl))
+                    freecamPosition.Z -= FreecamMoveSpeed;
+            }
+            else
+            {
+                bool fw = Foster.Framework.Input.Keyboard.Down(Keys.W);
+                bool rg = Foster.Framework.Input.Keyboard.Down(Keys.A);
+                bool bw = Foster.Framework.Input.Keyboard.Down(Keys.S);
+                bool lf = Foster.Framework.Input.Keyboard.Down(Keys.D);
```

Comment:
I don't like this. Why are the keys hardcoded here?
Just lake a new conntrol under `TASControls` for this.